examples/blaze/src/main/scala/com/example/http4s/blaze/demo/server/endpoints/JsonXmlHttpEndpoint.scala

```diff
@@ -0,0 +1,47 @@
+package com.example.http4s.blaze.demo.server.endpoints
+
+import cats.effect.Effect
+import io.circe.generic.auto._
+import org.http4s._
+import org.http4s.circe._
+import org.http4s.dsl.Http4sDsl
+
+import scala.xml._
+
+// Docs: http://http4s.org/v0.18/entity/
+class JsonXmlHttpEndpoint[F[_]](implicit F: Effect[F]) extends Http4sDsl[F] {
+
+  case class Person(name: String, age: Int)
+
+  /**
+    * XML Example for Person:
+    *
+    * <person>
+    *   <name>gvolpe</name>
+    *   <age>30</age>
+    * </person>
+    * */
+  object Person {
+    def fromXml(elem: Elem): Person = {
+      val name = (elem \\ "name").text
+      val age  = (elem \\ "age").text
+      Person(name, age.toInt)
+    }
+  }
+
+  def personXmlDecoder: EntityDecoder[F, Person] =
+    org.http4s.scalaxml.xml[F].map(Person.fromXml)
+
+  implicit def jsonXmlDecoder: EntityDecoder[F, Person] = jsonOf[F, Person] orElse personXmlDecoder
+
+  val service: HttpService[F] = HttpService {
+    case GET -> Root / ApiVersion / "media" =>
+      Ok("Send either json or xml via POST method. Eg: \n{\n  \"name\": \"gvolpe\",\n  \"age\": 30\n}\n or \n <person>\n  <name>gvolpe</name>\n  <age>30</age>\n</person>")
+
+    case req @ POST -> Root / ApiVersion / "media" =>
+      req.decode[Person] { person =>
```

Comment:
How many people are doing this vs. `req.as[Person].flatMap { person => ... }`?  There are two ways to do it, and I don't really know what's more popular in the wild.

Reply:
As an old Http4s user I'm used to this. I think unification of these two methods would be great to avoid confusion.

Reply:
`flatMap` seems natural.  I'm open to deprecation on the other if nobody can rise to a stirring defense of it.

Reply:
Fair enough. I'll change it in favor of `as`.